drivers/counter/counter_nrfx_timer.c

```diff
@@ -13,6 +13,8 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL);
 
 #define TIMER_CLOCK 16000000
 
+#define MS2TICKS(ms, prescaler) (((ms) * TIMER_CLOCK/(1 << (prescaler))) / 1000)
```

Comment:
I think defining additionally:
```
#define TIMER_FREQUENCY(idx) \
	(TIMER_CLOCK / (1U << DT_NORDIC_NRF_TIMER_TIMER_##idx##_PRESCALER))
```
and using it inside this macro as well as in the initialization of `.info.freq` could help in comprehension of the whole thing.